src/eureka/S3_data_reduction/nirspec.py

```diff
@@ -155,3 +155,201 @@ def fit_bg(dataim, datamask, n, meta, isplots=0):
+                the trace to the middle of the detector and make it straight
+                Shape: (Nb_columns)
+
+
+    '''
+
+    # get the dimensions of the data
+    nb_rows = data.shape[0]
+
+    # define an array of pixel positions (in their centers)
+    pix_centers = np.arange(nb_rows) + 0.5
+
+    # Compute the center of mass of each column and convert to integer (pixels)
+    column_coms = np.sum(pix_centers[:, None]*data, axis=0) / np.sum(data, axis=0)
+    column_coms = np.around(column_coms).astype(int)
+
+    # # try using the max of each column
+    # column_max = np.argmax(data, axis=1)
+    # column_coms = column_max.astype(int)
+
+    # define the new center (where we will align the trace) in the middle of the detector
+    new_center = int(nb_rows/2)
+
+    # define an array containing the needed shift to bring the COMs to the
+    # new center for each column of each integration
+    shifts = new_center - column_coms
+
+    # ensure columns of zeros or nans are not moved and dont induce bugs
+    shifts[column_coms < 0] = 0
+    shifts[column_coms > nb_rows] = 0
+
+    return shifts, new_center
+
+
+def roll_columns(data, shifts):
+    '''
+    For each column of each integration, rolls the columns by the 
+    values specified in shifts 
+
+    Input:
+    data   : (array) sequence of images. Shape: (Nb_integrations x Nb_rows x Nb_cols)
+    shifts : (array) shift to apply to each column of each integration
+                        Shape: (Nb_integrations x Nb_columns)
+
+    Return:
+    rolled_data : (array) sequence of images with same shape as data but 
+                            with the shifts applied
+
+    '''
+
+    # init straight_data
+    rolled_data = np.zeros_like(data)
+    # loop over all images (integrations)
+    for i in range(len(data)):
+        # do the equivalent of 'np.roll' but with a different shift in each column
+        
+        arr = np.swapaxes(data[i], 0, -1)
+        all_idcs = np.ogrid[[slice(0, n) for n in arr.shape]]
+
+        # make the shifts positive
+        shifts_i = shifts[i]
+        shifts_i[shifts_i < 0] += arr.shape[-1]
+
+        # apply the shifts
+        all_idcs[-1] = all_idcs[-1] - shifts_i[:, np.newaxis]
+        result = arr[tuple(all_idcs)]
+        arr = np.swapaxes(result, -1, 0)
+
+        # store in straight_data
+        rolled_data[i] = arr
+
+    return rolled_data
+    
+
+def straighten_trace(data, meta, log):
+    '''
+    Takes a set of integrations with a curved trace and shifts the 
+    columns to bring the center of mass to the middle of the detector
+    (and straighten the trace)
+
+    The correction is made by whole pixels (i.e. no fractional pixel shifts)
+    The shifts to be applied are computed once from the median frame and then
+    applied to each integration in the timeseries
+
+    Inputs:
+    =======
+    data : Eureka! data object (Astraeus object)
+    meta : Eureka! meta object
```

Comment:
You can just copy-paste the Parameters documentation from another function - keeping the documentation uniform will help end users quickly skim the docs.